### DISCUSSION TITLE
Why does LTeX check all documents in the workspace, even not opened ones?

### DISCUSSION THREAD

**Discussion by cands** [valentjn/vscode-ltex#191] [Q&A] [answered]
On my system (macOS 10.14.6, VSCode 1.52.0, LTeX v8.1.1) all files in the workspace (also files not opened or edited in VSCode) are checked when LTeX is enabled. Is this how it is supposed to work? 

The problem is that I have some relatively large multi-root workspaces that contain hundreds of Markdown and LaTeX files, and the CPU load becomes high for an extended period of time until all documents have been checked. Besides, many of the documents are finalized and spell / grammar checking is not needed anymore. 

I don't see a setting that can affect this behavior, is there a solution already or can it be fixed in future updates? A setting to choose whether all documents in workspace, only the currently edited document, or all documents currently opened in VSCode, should be checked could be useful.

**Comment by valentjn** [comment, marked as answer]
This is not expected behavior. LT<sub>E</sub>X is supposed to check only opened files, except when running the [`LTeX: Check all documents in workspace`](https://valentjn.github.io/vscode-ltex/docs/commands.html#ltex-check-all-documents-in-workspace) command. So this is a bug either in VS Code or in LT<sub>E</sub>X. [Please open a bug report with all the relevant info.](https://valentjn.github.io/vscode-ltex/docs/contributing-code-issues.html#how-to-report-bugs)

**Reply by cands** [reply]
Thank you for confirming unexpected behavior. I have now opened a bug report #197
